lm-agent/lm_agent/services/clean_jobs_and_bookings.py

```diff
@@ -117,10 +117,12 @@ async def clean_jobs_without_bookings(cluster_jobs: List[JobSchema]) -> List[Job
     logger.debug("##### Cleaning jobs without bookings")
 
     jobs_to_delete_call = []
+    jobs_to_delete = []
 
     for job in cluster_jobs[:]:
         if not job.bookings:
             jobs_to_delete_call.append(remove_job_by_slurm_job_id(job.slurm_job_id))
+            jobs_to_delete.append(job.slurm_job_id)
```

Comment:
Instead of having two lists here, why not accumulate the slurm_job_id values in a list and then use a comprehension to produce the list of async calls to make? Something like:

```
    jobs_to_delete[]
    for job in cluster_jobs[:]:
        if not job.bookings:
            jobs_to_delete.append(job.slurm_job_id)
            cluster_jobs.remove(job)
    ...     
    await asyncio.gather(*[remove_job_by_slurm_job_id(jid) for jid in jobs_to_delete])
    ```